batch/batch/worker/worker.py

```diff
@@ -1756,13 +1746,20 @@ def __init__(
             )
 
         self.containers = containers
+        self.monitors = {
+            task_name: c.new_resource_usage_monitor(self.resource_usage_path(task_name))
+            for task_name, c in containers.items()
+        }
```

Comment:
I prefer we do not lift monitors up to the Job level. This feels like a violation of the abstraction boundary. The container should be responsible for monitoring its resources and returning the results. It does seem worthwhile for a container to hold on to the resource monitor class and use that to `read` the file.